nomos-services/chain-service/src/lib.rs

```diff
@@ -1209,88 +1175,72 @@ where
     }
 
     #[expect(clippy::allow_attributes_without_reason)]
-    #[instrument(level = "debug", skip(tx_selector, blob_selector, relays))]
+    #[instrument(level = "debug", skip(tx_selector, relays))]
     async fn propose_block(
         parent: HeaderId,
         slot: Slot,
         proof: Risc0LeaderProof,
         tx_selector: TxS,
-        blob_selector: BS,
         relays: &CryptarchiaConsensusRelays<
             BlendService,
-            BS,
             ClPool,
             ClPoolAdapter,
-            DaPool,
-            DaPoolAdapter,
             NetAdapter,
             SamplingBackend,
             Storage,
             TxS,
             RuntimeServiceId,
         >,
-    ) -> Option<Block<ClPool::Item, DaPool::Item>> {
+    ) -> Option<Block<ClPool::Item>> {
         let mut output = None;
-        let cl_txs = get_mempool_contents(relays.cl_mempool_relay().clone());
-        let da_certs = get_mempool_contents(relays.da_mempool_relay().clone());
-        let blobs_ids = get_sampled_blobs(relays.sampling_relay().clone());
-        match futures::join!(cl_txs, da_certs, blobs_ids) {
-            (Ok(cl_txs), Ok(da_blobs_info), Ok(blobs_ids)) => {
-                let block = BlockBuilder::new(
-                    tx_selector,
-                    blob_selector,
-                    Builder::new(parent, slot, proof),
-                )
-                .with_transactions(cl_txs)
-                .with_blobs_info(
-                    da_blobs_info.filter(move |info| blobs_ids.contains(&info.blob_id())),
-                )
-                .build()
-                .expect("Proposal block should always succeed to be built");
+        let txs = get_mempool_contents(relays.cl_mempool_relay().clone()).await;
+        match txs {
+            Ok(txs) => {
+                let txs = tx_selector.select_tx_from(txs).collect::<Vec<_>>();
```

Comment:
Is the idea here to move the sampling result checking into the tx_selector, or should we continue appending to txs in this block by directly requesting a list of sampled blob IDs, like we did before?

Reply:
Ah, good catch, I see what you mean. Let's keep the second not to add additional stuff in here, we can think about ways to improve it later